src/textual/layouts/horizontal.py

```diff
@@ -36,7 +38,7 @@ def arrange(
                     ]
                 )
                 + (box_margins[0].left + box_margins[-1].right),
-                max(
+                min(
```

Comment:
Same comment re: test

Reply:
It is covered by a snapshot, but it probably does deserve a more specific test. Going to leave it for now though.